aiida_quantumespresso/calculations/pw.py

```diff
@@ -152,6 +155,38 @@ def define(cls, spec):
                     'is `False` and/or `electron_maxstep` is 0.')
         # yapf: enable
 
+    @staticmethod
+    def validate_inputs(value, _):
+        """Validate the top level namespace.
+
+        1. Check that the restart input parameters are set correctly. In case of 'nscf' and 'bands' calculations, this
+        means that ``parent_folder`` is provided, ``startingpot`` is set to 'file' and ``restart_mode`` is
+        'from_scratch'. For other calculations, if the ``parent_folder`` is provided, the restart settings must be set
+        to use some of the outputs.
+        """
+        parameters = value['parameters'].get_dict()
+        calculation_type = parameters.get('CONTROL', {}).get('calculation', 'scf')
+
+        # Check that the restart input parameters are set correctly
+        if calculation_type in ('nscf', 'bands'):
+            if 'parent_folder' not in value:
+                return f'`parent_folder` not provided for `{calculation_type}` calculation.'
+            if parameters.get('ELECTRONS', {}).get('startingpot', 'file') != 'file':
+                return f'`startingpot` should be set to `file` for a `{calculation_type}` calculation.'
+            if parameters.get('CONTROL', {}).get('CONTROL', None) == 'restart':
+                return f'`restart_mode` should be set to `from_scratch` for a `{calculation_type}` calculation.'
+        else:
+            if 'parent_folder' in value:
+                if not any([
+                    parameters.get('CONTROL', {}).get('CONTROL', None) == 'restart' and
+                    parameters.get('ELECTRONS', {}).get('startingpot', None) == 'file' and
+                    parameters.get('ELECTRONS', {}).get('startingwfc', None) == 'file'
+                ]):
```

Comment:
Don't think this does what you mean it too. Why make a list if you will get only a single boolean because of the `and` operators? Did you mean to use comma's instead of `and`? Or you just remove the `not any([])` from the conditional.

Reply:
Whoops 😅! I wanted to have commas indeed. Basically, if none of these are set for calculation that isn't `nscf` or `bands`, providing a `parent_folder` will have no effect, since QE's defaults are not set to restart in this case.